packages/js/src/server/integrations/unhandled_rejection_monitor.ts

```diff
@@ -0,0 +1,42 @@
+import Client from '../../server'
+import { fatallyLogAndExit } from '../util'
+import { Types } from '@honeybadger-io/core'
+
+export default class UnhandledRejectionMonitor {
+  protected __isReporting: boolean
+  protected __client: typeof Client
+
+  constructor(client: typeof Client) {
+    this.__isReporting = false
+    this.__client = client
+  }
+
+  /**
+   * If there are no other unhandledRejection listeners,
+   * we want to report the exception to Honeybadger and
+   * mimic the default behavior of NodeJs,
+   * which is to exit the process with code 1
+   */
+  hasOtherUnhandledRejectionListeners() {
+    return process.listeners('unhandledRejection').length > 1
+  }
+
+  handleUnhandledRejection(reason: unknown, _promise: Promise<unknown>) {
```

Comment:
Is `_promise` not being used here?

Reply:
See https://github.com/honeybadger-io/honeybadger-js/pull/1166/files#r1300209911